File: platform/android/app/src/main/java/com/snesonline/UiStyle.java
package com.snesonline;

public final class UiStyle {
    // Classic SNES palette (Android-only).
    // Primary Base, Console Light Grey, #CEC9CC
    // Secondary Base, Console Dark Grey, #908A99
    // Accent 1, Light Lavender (Buttons), #B5B6E4
    // Accent 2, Dark Purple (Buttons), #4F43AE
    // Text/Detail, Deep Charcoal, #211A21
    public static final int PRIMARY_BASE = 0xFFCEC9CC;
    public static final int SECONDARY_BASE = 0xFF908A99;
    public static final int ACCENT_1 = 0xFFB5B6E4;
    public static final int ACCENT_2 = 0xFF4F43AE;
    public static final int TEXT_DETAIL = 0xFF211A21;

    private UiStyle() {}

    public static int configBackground() { return 0xFFFFFFFF; }
    public static int configText() { return TEXT_DETAIL; }

    // Use slightly translucent dark overlay for in-game dialogs.
    public static int stateDialogBackground() { return 0xF2211A21; }
    public static int saveHeader() { return ACCENT_1; }
    public static int loadHeader() { return ACCENT_2; }
    public static int saveListBackground() { return 0xE6908A99; }
    public static int loadListBackground() { return 0xE6908A99; }
    public static int dividerColor() { return ACCENT_2; }
}

File: platform/ios/app/Sources/UiStyle.swift
import UIKit

// Mirrors Android's `com.snesonline.UiStyle`.
// Keep these values in sync so iOS and Android feel identical.
enum UiStyle {
    // Classic SNES palette
    // Primary Base, Console Light Grey, #CEC9CC
    // Secondary Base, Console Dark Grey, #908A99
    // Accent 1, Light Lavender (Buttons), #B5B6E4
    // Accent 2, Dark Purple (Buttons), #4F43AE
    // Text/Detail, Deep Charcoal, #211A21
    static let primaryBase = UIColor(hexRGB: 0xCEC9CC)
    static let secondaryBase = UIColor(hexRGB: 0x908A99)
    static let accent1 = UIColor(hexRGB: 0xB5B6E4)
    static let accent2 = UIColor(hexRGB: 0x4F43AE)
    static let textDetail = UIColor(hexRGB: 0x211A21)

    static var configBackground: UIColor { .white }
    static var configText: UIColor { textDetail }

    // In-game dialogs
    static var stateDialogBackground: UIColor { UIColor(hexRGBA: 0xF2211A21) }
    static var saveHeader: UIColor { accent1 }
    static var loadHeader: UIColor { accent2 }
    static var saveListBackground: UIColor { UIColor(hexRGBA: 0xE6908A99) }
    static var loadListBackground: UIColor { UIColor(hexRGBA: 0xE6908A99) }
    static var dividerColor: UIColor { accent2 }

    // Interactive theming (mirrors ConfigActivity.java)
    static var enabledButtonBg: UIColor { accent2 }
    static var disabledButtonBg: UIColor { UIColor(hexRGB: 0xE6E3E5) }
    static var enabledButtonText: UIColor { .white }
    static var disabledButtonText: UIColor { primaryBase }

    static var switchChecked: UIColor { accent2 }
    static var switchUnchecked: UIColor { secondaryBase }
    static var switchTrackChecked: UIColor { accent2.withAlphaComponent(0.40) }
    static var switchTrackUnchecked: UIColor { secondaryBase.withAlphaComponent(0.33) }
}

extension UIColor {
    convenience init(hexRGB: UInt32) {
        let r = CGFloat((hexRGB >> 16) & 0xFF) / 255.0
        let g = CGFloat((hexRGB >> 8) & 0xFF) / 255.0
        let b = CGFloat(hexRGB & 0xFF) / 255.0
        self.init(red: r, green: g, blue: b, alpha: 1.0)
    }

    convenience init(hexRGBA: UInt32) {
        let a = CGFloat((hexRGBA >> 24) & 0xFF) / 255.0
        let r = CGFloat((hexRGBA >> 16) & 0xFF) / 255.0
        let g = CGFloat((hexRGBA >> 8) & 0xFF) / 255.0
        let b = CGFloat(hexRGBA & 0xFF) / 255.0
        self.init(red: r, green: g, blue: b, alpha: a)
    }
}
